src/parsers/gene_association_file.py

```diff
@@ -0,0 +1,301 @@
+        "Assigned_By": "primary_knowledge_source"
+    }, inplace=True)
+    
+    try:
+        go_annotations_df["annotation_date"] = pd.to_datetime(go_annotations_df["annotation_date"], format='%Y%m%d').dt.strftime('%Y-%m-%d')
+    except Exception as e:
+        raise ValueError(f"Error parsing annotation_date: {e}")
+    
+    # Ensure the date is in string format
+    go_annotations_df["annotation_date"] = go_annotations_df["annotation_date"].astype(str)
+
+    # Add metadata columns
+    go_annotations_df["aggregator"] = "UniProt"
+    go_annotations_df["protocol_id"] = None
+
+    # Process multi-value fields
+    go_annotations_df["publications"] = go_annotations_df["publications"].apply(lambda x: str(x).split('|') if pd.notna(x) else None)
+    go_annotations_df["supporting_objects"] = go_annotations_df["supporting_objects"].apply(lambda x: str(x).split('|') if pd.notna(x) else None)
+
+    # Detect negation
+    go_annotations_df["negated"] = go_annotations_df["predicate"].str.contains(r"^NOT\|", regex=True)
+    go_annotations_df["predicate"] = go_annotations_df["predicate"].str.replace(r"^NOT\|", "", regex=True)
+
+    # Validate allowed predicates
+    invalid_predicates = ~go_annotations_df["predicate"].isin(ALLOWED_PREDICATES)
+    if invalid_predicates.any():
+        raise ValueError(f"Invalid predicates found: {go_annotations_df.loc[invalid_predicates, 'predicate'].unique()}")
+
+    # Merge with evidence types
+    merged = merge_evidence_mapping(go_annotations_df, eco_mapping_df)
+
+    # --- Fix multi-value columns ---
+    for col in ["publications", "supporting_objects"]:
+        if col in merged.columns:
+            merged[col] = merged[col].apply(lambda x: '|'.join(x) if isinstance(x, list) else x)
+
+    # --- Fix subject column ---
+    if "subject" in merged.columns:
+        merged["subject"] = merged["subject"].astype(str)
+
+    # Validate schema
+    validate_annotation_schema(merged)
+
+    # Save
+    merged = merged[ASSOCIATION_COL_HEADERS]  
+    merged.to_csv(output_path, index=False)
+
+
+# --- CLI Interface ---
+
+@click.command()
+@click.option('--input', '-i', required=True, help='Path to input GO annotation CSV file.')
+@click.option('--output', '-o', required=True, help='Path to output normalized CSV file.')
+
+def main(input, output):
+    """CLI entry point to process GO annotations."""
+
+    # Check input file exists
+    if not os.path.isfile(input):
+        print(f"Error: Input file '{input}' does not exist.", file=sys.stderr)
+        sys.exit(1)
+
+    # Check output directory exists
+    output_dir = os.path.dirname(output) or '.'
+    if not os.path.isdir(output_dir):
+        print(f"Error: Output directory '{output_dir}' does not exist.", file=sys.stderr)
+        sys.exit(1)
+
+    # Check ECO mapping file loads
+    eco_mapping_url = "http://purl.obolibrary.org/obo/eco/gaf-eco-mapping.txt"
+    try:
+        case = pd.read_csv(eco_mapping_url, sep="\t", comment="#", header=None)
```

Comment:
Since you're loading the gaf-eco-mapping file here, you may as well use it with `process_go_annotations(input, output, gaf_eco_mapping)` so that you don't have to load it again.